windows/utils/win_install_online_updates.yml

```diff
@@ -0,0 +1,84 @@
+#   win_updates_log_file: Windows Updates install log file path.
+#
+- name: "Set fact of Windows Updates install log path and initialize online update status"
+  ansible.builtin.set_fact:
+    win_updates_log_file: "C:\\win_updates_log.txt"
+    win_online_updates_succeed: false
+
+- name: "Get the list of available Windows Updates"
+  ansible.windows.win_updates:
+    server_selection: "windows_update"
+    category_names: "{{ win_updates_category_names | default('*') }}"
+    log_path: "{{ win_updates_log_file }}"
+    skip_optional: true
+    state: "searched"
+  delegate_to: "{{ vm_guest_ip }}"
+  register: win_updates_list
+  ignore_errors: "{{ win_updates_ignore_errors | default(false) }}"
+- name: "Print the list of Windows Updates"
+  ansible.builtin.debug: var=win_updates_list
+
+- name: "Install Windows Updates when updates found"
+  when:
+    - win_updates_list.found_update_count is defined
+    - win_updates_list.found_update_count | int != 0
+    - win_updates_list.updates is defined
+    - win_updates_list.updates | length != 0
+  block:
+    - name: "Install Windows Updates"
+      ansible.windows.win_updates:
+        server_selection: "windows_update"
+        category_names: "{{ win_updates_category_names | default('*') }}"
+        reject_list: "{{ win_updates_reject_list | default('') }}"
+        log_path: "{{ win_updates_log_file }}"
+        skip_optional: true
+        state: "installed"
+        reboot: true
+        reboot_timeout: 2700
+      delegate_to: "{{ vm_guest_ip }}"
+      register: win_updates_install_result
+      ignore_errors: "{{ win_updates_ignore_errors | default(false) }}"
+    - name: "Print Windows Updates install result"
+      debug: var=win_updates_install_result
+
+- name: "No need to install Windows Updates"
+  ansible.builtin.debug:
+    msg: "Will not execute Windows Updates installation task due to no update found."
+  when: >
+    (win_updates_list.found_update_count is undefined) or
+    (win_updates_list.found_update_count | int == 0)
+
+- name: "Set Windows Updates installation status"
+  ansible.builtin.set_fact:
+    win_online_updates_succeed: |-
+      {%- if win_updates_list.found_update_count is undefined or win_updates_list.found_update_count | int == 0 -%}
+        true
+      {%- elif win_updates_install_result is defined and win_updates_install_result.installed_update_count is defined and 
+        win_updates_install_result.found_update_count | int == win_updates_install_result.installed_update_count | int -%}
+        true
+      {%- else -%}
+        false
+      {%- endif -%}
+
+- name: "Check Windows Updates installation status"
+  ansible.builtin.assert:
+    that:
+      - win_online_updates_succeed
+    fail_msg: >-
+      Installed update count '{{ win_updates_install_result.installed_update_count | default(0) }}',
+      is not match found update count '{{ win_updates_install_result.found_update_count | default(0) }}'.
+  when: not win_updates_ignore_errors | bool
```

Comment:
Missing when for `win_updates_ignore_errors is undefined`. And Line 54 can be updated to `true`.

Reply:
Yes. Added it. Thanks.